plugins/tradingAdvisor/baseTradingMethod.js

```diff
@@ -69,6 +68,11 @@ var Base = function(settings) {
   // let's run the implemented starting point
   this.init();
 
+  //if no requiredHistory was provided, set default from tradingAdvisor
+  if (!this.requiredHistory){
```

Comment:
Would you mind checking if it's actually undefined? This current check would also reset it if the strategy has defined it as `0`. A simple way would be to use lodash to check if requiredHistory is a number (`_.isNumber`).